presto-orc/src/main/java/com/facebook/presto/orc/metadata/RowGroupBloomfilter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata;
+
+import org.apache.hadoop.hive.ql.io.orc.OrcProto;
+import org.apache.hive.common.util.BloomFilter;
+
+import static java.util.Objects.requireNonNull;
+
+public final class RowGroupBloomfilter
+{
+    private final OrcProto.BloomFilter orcBloomfilter;
+    private HiveBloomFilter bloomfilter;
+
+    public RowGroupBloomfilter(OrcProto.BloomFilter orcBloomfilter)
+    {
+        this.orcBloomfilter = requireNonNull(orcBloomfilter, "orcBloomfilter is null");
+    }
+
+    public OrcProto.BloomFilter getOrcBloomfilter()
+    {
+        return orcBloomfilter;
+    }
+
+    public BloomFilter getBloomfilter()
+    {
+        if (bloomfilter == null) {
+            // lazy initialise the bloomfilter, we can only use it in a limited set of scenarios
+            bloomfilter = new HiveBloomFilter(getOrcBloomfilter().getBitsetList(), getOrcBloomfilter().getBitsetCount() * 64, getOrcBloomfilter().getNumHashFunctions());
```

Comment:
Are you sure this call is expensive?  My guess is doing this eagerly is more efficient because it because the code would be retaining a `long[]` instead of a `List<Long>`.  Unless you are confident this is a big win, do it eagerly.